packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.tsx

```diff
@@ -190,11 +193,17 @@ export class AtomicCategoryFacet
     };
     this.facet = buildCategoryFacet(this.bindings.engine, {options});
     this.facetId = this.facet.state.facetId;
-    registerFacetToStore(this.bindings.store, 'categoryFacets', {
-      label: this.label,
-      facetId: this.facetId!,
-      element: this.host,
-    });
+    registerFacetToStore(
+      this.bindings.store,
+      'categoryFacets',
+      {
+        label: this.label,
+        facetId: this.facetId!,
+        facet: this.facet,
```

Comment:
We now need to pass the facet directly, so we can call `enable`/`disable` on it.

At first, I added this because I also wanted to read the facet's current state, but this is no longer needed since I changed my mind and decided to read the facet request state on the engine instead of the facet controller state.

If you prefer, I can remove this requirement and instead dispatch the `enable`/`disable` actions directly.

If you want to know why I decided to read the facet request state instead of the facet controller state, it's because the state of the url manager corresponds with the state of the request, not the state of the response, so it's much easier to keep in sync this way.